src/Account.sol

```diff
@@ -73,70 +66,45 @@ contract Account is IAccount, OpsReady, Auth, Initializable {
-
-    /// @notice helpful modifier to check non-zero values
-    /// @param value: value to check if zero
-    modifier notZero(uint256 value, bytes32 valueName) {
-        if (value == 0) revert ValueCannotBeZero(valueName);
-
-        _;
-    }
+    mapping(uint256 id => ConditionalOrder order) internal conditionalOrders;
 
     /*//////////////////////////////////////////////////////////////
                               CONSTRUCTOR
     //////////////////////////////////////////////////////////////*/
 
     /// @notice disable initializers on initial contract deployment
     /// @dev set owner of implementation to zero address
-    /// @param addressResolver: address of Synthetix ReadProxyAddressResolver
-    /// @param marginAsset: address of Synthetix ProxyERC20sUSD
-    /// @param gelato: address of Gelato
-    /// @param ops: address of Ops
+    /// @param _factory: address of the Smart Margin Account Factory
+    /// @param _events: address of the contract used by all accounts for emitting events
+    /// @param _marginAsset: address of the Synthetix ProxyERC20sUSD contract used as the margin asset
+    /// @param _futuresMarketManager: address of the Synthetix FuturesMarketManager
+    /// @param _gelato: address of Gelato
+    /// @param _ops: address of Ops
     constructor(
-        address addressResolver,
-        address marginAsset,
-        address gelato,
-        address ops
-    ) Auth(address(0)) OpsReady(gelato, ops) {
+        address _factory,
+        address _events,
+        address _marginAsset,
+        address _futuresMarketManager,
+        address _systemStatus,
+        address _gelato,
+        address _ops
+    ) Auth(address(0)) OpsReady(_gelato, _ops) {
         // recommended to use this to lock implementation contracts
         // that are designed to be called through proxies
         _disableInitializers();
 
-        ADDRESS_RESOLVER = IAddressResolver(addressResolver);
-        MARGIN_ASSET = IERC20(marginAsset);
+        FACTORY = IFactory(_factory);
+        EVENTS = IEvents(_events);
+        MARGIN_ASSET = IERC20(_marginAsset);
+        FUTURES_MARKET_MANAGER = IFuturesMarketManager(_futuresMarketManager);
+        SYSTEM_STATUS = ISystemStatus(_systemStatus);
     }
 
-    /// @notice initialize contract (only once) and transfer ownership to specified address
-    /// @dev ensure resolver and sUSD addresses are set to their proxies and not implementations
+    /// @notice initialize contract (only once), transfer ownership to specified address
     /// @param _owner: account owner
-    /// @param _settings: contract address for account settings
-    /// @param _events: address of events contract for accounts
-    /// @param _factory: contract address for account factory
-    function initialize(
-        address _owner,
-        address _settings,
-        address _events,
-        address _factory
-    ) external initializer {
+    function initialize(address _owner) external initializer {
```

Comment:
**Suggestion/thought:** Now that we all we have in the the initializer is a transfer of Ownership, we could actually get rid of the initializer all together and instead just initially set the owner to `msg.sender` in `Account.constructor`, so the implementation is initially owned by the deployer. Then you could call `Account.transferOwnership` via the proxy in `Factory.newAccount`. Then we could get rid of the `Initializable` dependency entirely.
If you want the implementation to be owned by `address(0)` that could then be achieved by calling `Account.transferOwnership` directly in the deploy script.